File: Algorithm/Code_Materials/python_source/charpter4_stack/level2/min_stack.py
class MinStack:
    def __init__(self):
        self.xStack = []
        self.minStack = []
        self.minStack.append(float('inf'))

    def push(self, x):
        self.xStack.append(x)
        self.minStack.append(min(self.minStack[-1], x))

    def pop(self):
        self.xStack.pop()
        self.minStack.pop()

    def top(self):
        return self.xStack[-1]

    def getMin(self):
        return self.minStack[-1]

File: Algorithm/Leetcode/Code_Materials/c_or_cpp/charpter4_stack/level2/MinStack.cpp
#include <iostream>
#include <stack>
#include <climits>

using namespace std;

class MinStack {
    stack<int> x_stack;
    stack<int> min_stack;
public:
    MinStack() {
        min_stack.push(INT_MAX);
    }
    
    void push(int x) {
        x_stack.push(x);
        min_stack.push(min(min_stack.top(), x));
    }
    
    void pop() {
       x_stack.pop();
       min_stack.pop();
    }
    
    int top() {
       return x_stack.top();
    }
    
    int getMin() {
        return min_stack.top();
    }
};

int main() {
   // 创建一个MinStack对象
   MinStack minStack;

   // 进行一些操作
   minStack.push(4);
   minStack.push(2);
   minStack.push(6);
   minStack.push(1);

   cout << "top: " << minStack.top() << endl; // 应该输出1
   cout << "getMin: " << minStack.getMin() << endl; // 应该输出1

   minStack.pop();
   cout << "top: " << minStack.top() << endl; // 应该输出6
   cout << "getMin: " << minStack.getMin() << endl; // 应该输出2

   return 0;
}